test/e2e/remote-api.test.js

```diff
@@ -0,0 +1,247 @@
+/* global ipfs, webuiUrl, page, describe, it, expect, beforeAll */
+
+const { createController } = require('ipfsd-ctl')
+const getPort = require('get-port')
+const http = require('http')
+const httpProxy = require('http-proxy')
+const basicAuth = require('basic-auth')
+const toUri = require('multiaddr-to-uri')
+
+// ipfs is a global 'local' node used for other tests,
+// we reuse it for testing the most basic setup with no auth
+const localIpfs = ipfs
+
+// Basic Auth Proxy Setup
+// -----------------------------------
+// Why do we support and test Basic Auth?
+// Some users choose to access remote API.
+// It requires setting up reverse proxy with correct CORS  and Basic Auth headers,
+// but when done properly, should work. This test sets up a proxy which
+// acts as properly protected and configured remote API to ensure there are no
+// regressions for this difficult to test use case.
+let ipfsd
+let proxyd
+let user
+let password
+let proxyPort
+let startProxyServer
+
+beforeAll(async () => {
+  await page.goto(webuiUrl)
+  // spawn an ephemeral local node to ensure we connect to a different, remote node
+  ipfsd = await createController({
+    type: 'go',
+    ipfsBin: require('go-ipfs').path(),
+    ipfsHttpModule: require('ipfs-http-client'),
+    test: true,
+    disposable: true
+  })
+
+  // set up proxy in front of remote API to provide CORS and Basic Auth
+  user = 'user'
+  password = 'pass'
+
+  const proxy = httpProxy.createProxyServer()
+  const remoteApiUrl = toUri(ipfsd.apiAddr.toString(), { assumeHttp: true })
+  proxy.on('proxyReq', (proxyReq, req, res, options) => {
+    // swap Origin before passing to the real API
+    // This way internal check of go-ipfs does get triggered (403 Forbidden on Origin mismatch)
+    proxyReq.setHeader('Origin', remoteApiUrl)
+    proxyReq.setHeader('Referer', remoteApiUrl)
+    proxyReq.setHeader('Host', new URL(remoteApiUrl).host)
+  })
+
+  proxy.on('error', function (err, req, res) {
+    res.writeHead(500, { 'Content-Type': 'text/plain' })
+    res.end(`proxyd error: ${JSON.stringify(err)}`)
+  })
+
+  proxyd = http.createServer((req, res) => {
+    // console.log(`${req.method}\t\t${req.url}`)
```

Comment:
You forgot to remove this? :)

Reply:
no, left in on purpose in case someone wants to debug this and is unsure where to add console log (there is server, then proxy inside of server, then we override stuff...) ;-)